### PR TITLE
DEV-683 Refactor Meme Factory Contract

### PR DESCRIPTION
**Pull request by iHiteshAgrawal**
## Summary by CodeRabbit

- **New Features**
	- Introduced new error types and events for better transaction tracking and error handling.
	- Added new function signatures and structs to enhance trading capabilities.
	- Implemented minimum liquidity requirements and lock duration constraints for token creation.
	- Introduced a launch fee for deploying new tokens.
- **Bug Fixes**
	- Adjusted function signatures and struct formats for consistency and readability.
- **Refactor**
	- Refactored various function declarations for clarity.
	- Updated test cases to support new features and parameters.
- **Chores**
	- Updated deployment scripts with new parameters and configuration adjustments.
	- Modified project configuration for network setups.
- **Documentation**
	- Added comprehensive pool state information retrieval methods to the test interface documentation.

### CODE REVIEW

File: contracts/MemeFactory.sol [hunk 1]
```diff
@@ -96,50 +97,64 @@ contract MemeFactory is Ownable {
...
+        address _vaporDexAdapter;
+        address _usdc;
+        address _vape;
+        uint256 _launchFee;
+        uint256 _minLiquidityETH;
+        uint40 _minLockDuration;
+        address _sablier;
+        address _nonFungiblePositionManager;
+        address _teamMultisig;
+    }
```
Comment (mejiasd3v) [inline contracts/MemeFactory.sol]: nice addition, please remove the underscore, not needed for internal struct fields IMO

File: contracts/MemeFactory.sol [hunk 2]
```diff
@@ -159,12 +175,17 @@ contract MemeFactory is Ownable {
         string memory _symbol,
         uint256 _totalSupply,
         uint256 _tradingStartsAt,
+        uint40 lockDuration,
```
Comment (mejiasd3v) [inline contracts/MemeFactory.sol:181]: I think this should be an enum and not "any" numeric value, wdyt?

**Review by mejiasd3v — APPROVED** [review]
minor comments, looks good, feel free to apply suggestions

File: contracts/MemeFactory.sol (hunk 1 above)
Comment (iHiteshAgrawal) [reply contracts/MemeFactory.sol]: fixed

**Review by mejiasd3v — APPROVED** [review]

**Pull request merged**